### ISSUE TITLE
Step 1 parameter optimization failure

### ISSUE THREAD

**Issue by emilliman5** [GreenleafLab/NucleoATAC#16]
Hi Alicia,

I am getting the following error(below) when running nucleoatac on one of my ATAC datasets. I have successfully run nucleoatac with this bed file and a different ATAC bam. And I have successfully run `nucleoatac` with the bam file on a different bed file... I re-installed `nucleoatac` and set maxfev=1800 but have gotten the same error. I also tried increasing the bed file intervals, but to no avail.  Any ideas on a workaround would be very helpful. Let me know if you need more info.

Thanks!
Eric

`nucleoatac run --bed ../../../Features/scRNA_veh_oTSS_sort_200bpwin.bed --bam ../UL3_siH1e_NS50005_NS50031.bam --out scRNA_veh_oTSS_200bp_win_UL3_siH1e_NS50005_NS50031 --write_all --fasta ~/Documents/genome_files/hg19_chromfa.fa --cores 8`

Command run:  /root/anaconda/bin/nucleoatac run --bed ../../../Features/scRNA_veh_oTSS_sort_200bpwin.bed --bam ../UL3_siH1e_NS50005_NS50031.bam --out scRNA_veh_oTSS_200bp_win_UL3_siH1e_NS50005_NS50031 --write_all --fasta /home/millimanej/Documents/genome_files/hg19_chromfa.fa --cores 8
nucleoatac version 0.2.1
start run at: 2015-10-02 13:55
---------Step1: Computing Occupancy and Nucleosomal Insert Distribution---------
Traceback (most recent call last):
  File "/root/anaconda/bin/nucleoatac", line 291, in <module>
    main()
  File "/root/anaconda/bin/nucleoatac", line 75, in main
    run_occ(occ_args)
  File "/root/anaconda/lib/python2.7/site-packages/nucleoatac/run_occ.py", line 93, in run_occ
    fragment_dist.modelNFR()
  File "/root/anaconda/lib/python2.7/site-packages/nucleoatac/Occupancy.py", line 39, in modelNFR
    p0=p0)
  File "/root/anaconda/lib/python2.7/site-packages/scipy/optimize/minpack.py", line 560, in curve_fit
    raise RuntimeError(msg)
RuntimeError: Optimal parameters not found: Number of calls to function has reached maxfev = 600.

**Comment by AliciaSchep** [comment]
Hi Eric,

What does the fragment size distribution for that sample look like within those regions?  (You can get the fragment size distribution within your peak regions using the  command 'pyatac sizes --bam name-of-bam-file.bam --bed name-of-bed-file.bed' ).  Is the set of regions very small?

Thanks,
Alicia

**Comment by emilliman5** [comment]
Alicia,

The regions in the bed-file are all 550bp. 

![ul3_sih1e_ns50005_ns50031](https://cloud.githubusercontent.com/assets/7238842/10290615/27bb7456-6b73-11e5-955a-225986ee6b2f.png)

**Comment by AliciaSchep** [comment]
How does that compare to the global fragment size distribution? (e.g. as would be obtained by 'pyatac sizes --bam name-of-bam-file.bam')

I can see why the fit failed, that distribution is fairly different from what I would expect from typical sample.  I have some ideas on how to make fit more robust but still need to actually implement those.

**Comment by emilliman5** [comment]
Alicia, 

It looks Iike I will have wait until the next version... This sample and some of its brethren have had their chromatin altered quite drastically by depletion of various histone genes. I've attached the global fragment length distribution for your viewing pleasure.

Thanks for your input.
Eric
![ul3_sih1e_ns50005_ns50031](https://cloud.githubusercontent.com/assets/7238842/10317051/79f0268c-6c30-11e5-8c9d-4b851eb42223.png)

**Comment by AliciaSchep** [comment]
Hi Eric,
Any chance you could attach one of the text files with the fragment size information (*.fragmentsizes.txt)?  I'd like to test out a new fit function on one of those fragment size distributions.
thanks,
Alicia

**Comment by emilliman5** [comment]
Alicia,

I have put two sets of fragment files (4 total) into the examples dir in a fork of your repo emilliman5/NucleoATAC. They are my control and treatment data at TSSs and peaks called from the ATAC data. The treatment data are the ones that are not being fit, while the controls seem to run fine. 

thanks for the help,
Eric

**Comment by AliciaSchep** [comment]
Thanks for providing the files.  I have tested them with the new fit function and all seem to work without error and produce reasonable results.  I have merged the branch with that fit function into the master to create version 0.3.0.  That version will hopefully work for your samples.

![trt_atac_peaks fit](https://cloud.githubusercontent.com/assets/6809790/10655061/13243338-7826-11e5-9c3d-e212e91013de.png)
